src/modules/battery.rs

```diff
@@ -15,12 +15,30 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let config: BatteryConfig = BatteryConfig::try_load(module.config);
 
     // Parse config under `display`.
-    // Select the first style that match the threshold,
-    // if all thresholds are lower do not display battery module.
-    let display_style = config
-        .display
-        .iter()
-        .find(|display_style| percentage <= display_style.threshold as f32)?;
+    // Select the style that is most minimally greater than the current battery percentage.
+    // If no such style exists do not display battery module.
+    let mut closest_style = None;
+    for display_style in &config.display {
+        // Skip style if its threshold is lower than the current battery percentage.
+        if percentage > display_style.threshold as f32 {
+            continue;
+        }
+        // Unwrap the `Option`
+        let Some(closest_so_far) = closest_style else {
+            // If `closest_style` is `None` then this is the first valid style
+            // encountered, so save it and continue to the next loop iteration.
+            closest_style = Some(display_style);
+            continue;
+        };
+        // If this style's threshold is less than the previous closest so far,
+        // save it and continue to the next loop iteration
+        if display_style.threshold < closest_so_far.threshold {
+            closest_style = Some(display_style);
+            continue;
+        }
+    }
+    let display_style = closest_style?;
+    //let display_style = config.display.iter().find(|display_style| percentage <= display_style.threshold as f32)?;
```

Comment:
I think that it would be easier to handle it like this:
```suggestion
    let display_style = config
        .display
        .iter()
        .filter(|display_style| percentage <= display_style.threshold as f32)
        .min_by_key(|display_style| display_style.threshold)?;
```

Reply:
The Iterator API never cease to amaze me... I blame my lack of sleep for not looking through the Iterator docs a little more closely for this function. That is indeed a lot cleaner and more concise though, I'll update it later today when I have a break at work.